https://gcc.gnu.org/ml/gcc-patches/2015-08/msg00375.html

2015-08-06  Mike Frysinger  <vapier@gentoo.org>

Since the 3.0.3 release of gperf (made in May 2007), the generated func
has had the gnu_inline attribute applied to it.  The gcc source however
has not been updated to include that which has lead to a mismatch.


diff -Naur gcc-4.7.4-pure/gcc/cp/cfns.gperf gcc-4.7.4/gcc/cp/cfns.gperf
--- gcc-4.7.4-pure/gcc/cp/cfns.gperf	2009-04-21 20:03:23.000000000 +0100
+++ gcc-4.7.4/gcc/cp/cfns.gperf	2016-07-18 17:45:40.060690800 +0100
@@ -22,6 +22,9 @@
 static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
 #endif
 const char * libc_name_p (const char *, unsigned int);
 %}
diff -Naur gcc-4.7.4-pure/gcc/cp/cfns.h gcc-4.7.4/gcc/cp/cfns.h
--- gcc-4.7.4-pure/gcc/cp/cfns.h	2009-04-21 20:03:23.000000000 +0100
+++ gcc-4.7.4/gcc/cp/cfns.h	2016-07-18 17:45:40.098693000 +0100
@@ -53,6 +53,9 @@
 static unsigned int hash (const char *, unsigned int);
 #ifdef __GNUC__
 __inline
+#ifdef __GNUC_STDC_INLINE__
+__attribute__ ((__gnu_inline__))
+#endif
 #endif
 const char * libc_name_p (const char *, unsigned int);
 /* maximum key range = 391, duplicates = 0 */
